solr/core/src/test/org/apache/solr/search/TestThinCache.java

```diff
@@ -26,13 +26,63 @@
 import org.apache.solr.SolrTestCaseJ4;
 import org.apache.solr.metrics.SolrMetricManager;
 import org.apache.solr.metrics.SolrMetricsContext;
+import org.apache.solr.util.EmbeddedSolrServerTestRule;
+import org.apache.solr.util.TestHarness;
 import org.apache.solr.util.stats.MetricUtils;
 import org.junit.BeforeClass;
+import org.junit.ClassRule;
 import org.junit.Test;
 
 /** Test for {@link ThinCache}. */
 public class TestThinCache extends SolrTestCaseJ4 {
 
+  @ClassRule public static EmbeddedSolrServerTestRule solrRule = new EmbeddedSolrServerTestRule();
+  public static final String SOLR_NODE_LEVEL_CACHE_XML =
```

Comment:
String literal (vs a separate file) keeps the test with the config its testing.  Constant field allows it to be used in the "TestSolrXml" test too.
Looking forward to Java 17 where we could use multi-line String syntax!  (maybe worth requiring Java 17 for tests?)

Reply:
I like seeing xml next ot the test too, helps me understand what is going on.  I use the tests as a "reference" to figure things out a lot.